### PR TITLE
[SD-3386] map iptc subject to anpa category and vice-versa

### PR DESCRIPTION
**Pull request by marwoodandrew**
This PR is dependent on superdesk/superdesk-core#46

### CODE REVIEW

File: server/apps/io/update_ingest_tests.py [hunk 1]
```diff
@@ -50,9 +50,13 @@ def test_the_second_update_fails_if_already_running(self):
 
 
 class UpdateIngestTest(SuperdeskTestCase):
```
Comment (sivakuna-aap) [inline server/apps/io/update_ingest_tests.py]: Can we add one more test for checking if subject is empty when there's no mapping vocabulary and a mapping vocabulary with inactive entries?

**Comment by sivakuna-aap** [comment]
:+1: except the minor one.

File: server/apps/io/update_ingest_tests.py [hunk 2]
```diff
@@ -350,3 +351,93 @@ def test_files_dont_duplicate_ingest(self):
...
+        provider_service = self.provider_services[provider.get('type')]
+        provider_service.provider = provider
+
+        items = provider_service.parse_file(guid, provider)
+        service = get_resource_service('ingest')
+        service.post(items)
+
+        # ingest the items and check the subject code has been derived
+        self.ingest_items(items, provider)
+        self.assertTrue('subject' not in items[0])
```
Comment (plamut) [inline server/apps/io/update_ingest_tests.py]: **Tip:** There also exists the `self.assertNotIn()` method (but it's very minor, no need for a change just because of this).

File: server/apps/io/dpa.py [hunk 3]
```diff
@@ -55,3 +55,16 @@ def _update(self, provider):
             except Exception as ex:
                 self.move_file(self.path, filename, provider=provider, success=False)
                 raise ParserError.parseFileError('DPA', filename, ex, provider)
+
+    def parse_file(self, filename, provider):
```
Comment (plamut) [inline server/apps/io/dpa.py]: A basic docstring would be useful here, e.g. is `filename` a full or a relative path, etc.

**Comment by plamut** [comment]
Looks good, just please check the test coverage as @sivakuna-aap has already mentioned, just in case.

:+1:

**Comment by ioanpocol** [comment]
:+1:  (except comments)

File: server/features/steps/steps.py [hunk 4]
```diff
@@ -357,7 +357,7 @@ def fetch_from_provider(context, provider_name, guid, routing_scheme=None, desk_
     provider_service = context.provider_services[provider.get('type')]
     provider_service.provider = provider
 
-    if provider.get('type') == 'aap' or provider.get('type') == 'teletype':
+    if provider.get('type') == 'aap' or provider.get('type') == 'teletype' or provider.get('type') == 'dpa':
```
Comment (amagdas) [inline server/features/steps/steps.py]: It reads better like `if provider.get('type') in {'aap', 'teletype', 'dpa'}:`

**Comment by amagdas** [comment]
:+1:

**Comment by mdhaman** [comment]
:+1:  (except comments)

**Comment by sivakuna-aap** [comment]
@superdesk/superdesk-europe can someone having access to master.dev db merge this as it involves updating vocabularies collection?

**Comment by amagdas** [comment]
I don't think that manual deployment is needed for this, merged, if it is needed, will fix the DB.

**Comment by amagdas** [comment]
Update: vocabularies are updated, no need for manual deployment.
